compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -169,6 +156,594 @@ bool TR_CFGSimplifier::simplifyBooleanStore()
+       && !storeNode->getDataType().isAddress())
+      return false;
+
+   if (trace())
+      traceMsg(comp(), "   StoreNode data type checks out\n");
+
+   if (!diamond && !storeNode->getSymbolReference()->getSymbol()->isAutoOrParm())
+      return false;
+
+   if (trace())
+      traceMsg(comp(), "   StoreNode symRef checks out\n");
+
+   if (!performTransformation(comp(), "%sReplace conditional store with store of an appropriate ternary at node [%p]\n", OPT_DETAILS, compareNode))
+      return false;
+
+   _cfg->invalidateStructure();
+
+   TR::Node *select = TR::Node::create(storeNode, comp()->il.opCodeForTernarySelect(storeNode->getDataType()), 3);
+   select->setAndIncChild(0,
+      TR::Node::create(compareNode, compareNode->getOpCode().convertIfCmpToCmp(), 2,
+         compareNode->getFirstChild(), 
+         compareNode->getSecondChild()));
+   select->setAndIncChild(1,
+      trueValue ? (needToDuplicateTree ? trueValue->duplicateTree() : trueValue) : TR::Node::createWithSymRef(comp()->il.opCodeForDirectLoad(storeNode->getDataType()), 0, storeNode->getSymbolReference()));
+   select->setAndIncChild(2,
+      falseValue ? (needToDuplicateTree? falseValue->duplicateTree() : falseValue) : TR::Node::createWithSymRef(comp()->il.opCodeForDirectLoad(storeNode->getDataType()), 0, storeNode->getSymbolReference()));
+   TR::TreeTop *cmov = TR::TreeTop::create(comp(), TR::Node::createWithSymRef(storeNode, storeNode->getOpCodeValue(), 1, select, storeNode->getSymbolReference()));
+   compareTree->insertBefore(cmov);
+   if (isHeapificationStore)
+      cmov->getNode()->setHeapificationStore(true);
+
+   if (trace())
+      traceMsg(comp(), "End simplifySimpleStore. New ternary node is n%dn\n", select->getGlobalIndex());   
+
+   TR::Block *dest;
+   if (diamond) {
+      dest = toBlock(_next1->getSuccessors().front()->getTo());
+      _cfg->addEdge(_block, dest);
+      _cfg->removeEdge(_block, _next1);
+      _cfg->removeEdge(_block, _next2);
+   } else if (triangle2) {
+      dest = _next1;
+      _cfg->removeEdge(_block, _next2);
+   } else if (triangle1) {
+      dest = _next2;
+      _cfg->removeEdge(_block, _next1);
+   }
+   if (_block->getNextBlock() != dest)
+      {
+      TR::Node *gotoNode = TR::Node::create(compareNode, TR::Goto, 0);
+      gotoNode->setBranchDestination(dest->getEntry());
+      _block->append(TR::TreeTop::create(comp(), gotoNode));
+      }
+
+   TR::TransformUtil::removeTree(comp(), compareTree);
+   
+   TR::DebugCounter::incStaticDebugCounter(comp(), TR::DebugCounter::debugCounterName(comp(), "cfgSimpCMOV/%s/(%s)", (diamond ? "diamond" : (triangle1 ? "triangle1" : "triangle2")), comp()->signature()));
+   return true;
+   }
+
+bool OMR::CFGSimplifier::simplifyNullToException(bool needToDuplicateTree)
+   {
+   static char *disableSimplifyExplicitNULLTest = feGetEnv("TR_disableSimplifyExplicitNULLTest");
+   static char *disableSimplifyNullToException = feGetEnv("TR_disableSimplifyNullToException");
```

Comment:
Is there a reason for using an environment variable instead of an option here?

Reply:
Commented previously.